submissions/vien-peter/scripts/article.js

```diff
@@ -137,5 +138,38 @@
       };
     });
   };
+  Article.allYears = function() {
+
+    return Article.all.map(function(article) {
+      var date = new Date(article.publishedOn);
+      var fullYear = date.getFullYear();
+      return fullYear;
+    })
+    .reduce(function(acc, curr, idx, arr) {
+      if(acc.indexOf(curr) === -1) {
+        acc.push(curr);
+      }
+      return acc;
+    },[]);
+  };
+
+  Article.numWordsPerYear = function() {
+    return Article.allYears().map(function(a) {
+      return {
+        year: a,
+        numWordsPerYear: Article.all.filter(function(article) {
+          var date = new Date(article.publishedOn);
+          var fullYear = date.getFullYear();
+          return a === fullYear;
```

Comment:
You could probably abstract these three lines out to a helper function and DRY out your code.